neuraxle/metaopt/auto_ml.py

```diff
@@ -623,40 +638,66 @@ def _fit_data_container(self, data_container: DataContainer, context: ExecutionC
-            p = copy.deepcopy(self.pipeline)
+            auto_ml_data = AutoMLContainer(
+                trial_number=trial_number,
+                trials=self.hyperparams_repository.load_all_trials(TRIAL_STATUS.SUCCESS),
+                hyperparameter_space=self.pipeline.get_hyperparams_space(),
+                main_scoring_metric_name=self.trainer.get_main_metric_name()
+            )
 
             with self.hyperparams_repository.new_trial(auto_ml_data) as repo_trial:
-                try:
+                for training_data_container, validation_data_container in validation_splits:
+                    p = copy.deepcopy(self.pipeline)
                     p.update_hyperparams(repo_trial.hyperparams)
                     repo_trial.set_hyperparams(p.get_hyperparams())
 
-                    repo_trial = self.trainer.fit(
-                        p=p,
-                        train_data_container=training_data_container,
-                        validation_data_container=validation_data_container,
-                        trial=repo_trial,
-                        context=context
-                    )
-
-                    repo_trial.set_success()
-                    self.print_func('trial {}/{} score: {}'.format(trial_number + 1, self.n_trial, repo_trial.get_validation_scores()[-1]))
-                except Exception as error:
-                    track = traceback.format_exc()
-                    self.print_func(track)
-                    repo_trial.set_failed(error)
+                    with repo_trial.new_validation_split(p) as repo_trial_split:
+                        try:
+                            trial_split_description = '{}/{} split {}/{}\nhyperparams: {}\n'.format(
+                                trial_number + 1,
+                                self.n_trial,
+                                repo_trial_split.split_number + 1,
+                                len(training_data_container),
+                                json.dumps(repo_trial.hyperparams, sort_keys=True, indent=4),
+                                # TODO: make hyperparams space serializable
+                                # json.dumps(auto_ml_data.hyperparameter_space.to_flat_as_dict_primitive(), sort_keys=True, indent=4),
+                            )
+
+                            repo_trial_split = self.trainer.fit_trial_split(
+                                trial_split=repo_trial_split,
+                                train_data_container=training_data_container,
+                                validation_data_container=validation_data_container,
+                                context=context
+                            )
+
+                            repo_trial_split.set_success()
+
+                            self.print_func('{} score: {}'.format(
+                                trial_split_description,
+                                repo_trial_split.get_validation_score()
+                            ))
```

Comment:
Please move this print before `self.trainer.fit_trial_split` to allow the user to see the hyperparam samples before fitting if he wants to cancel it. If a print of the hyperparam samples is already made before the `self.trainer.fit_trial_split`, then I might still move this print above but without the score and also without the hp samples

Reply:
there is now 3 prints : fitting trial 1/1 split 1/1 ..., success trial 1/1 split 1/1, or failed trial 1/1 split 1/1...